python/mcap-protobuf-support/mcap_protobuf/decoder.py

```diff
@@ -59,6 +59,26 @@ def _add(fd: FileDescriptorProto):
 
             messages = GetMessageClassesForFiles([fd.name for fd in fds.file], pool)
 
+            # GetMessageClassesForFiles only fetches top-level message definitions in the file.
+            #
+            # We must traverse the top-level message definitions to recursively find and append any
+            # nested message definitions.
+            nested_messages = {}
+            for message_class in messages.values():
+                nested_descriptions = list(
+                    message_class.DESCRIPTOR.nested_types_by_name.values()
+                )
+                while nested_descriptions:
+                    nested_desc = nested_descriptions.pop()
+                    nested_messages[nested_desc.full_name] = GetMessageClass(
```

Comment:
why does this not mutate `messages` directly?

Reply:
We will otherwise be mutating it while we are iterating through the dictionary.